src/tools/blur/blurtool.cpp

```diff
@@ -16,73 +16,127 @@
 //     along with Flameshot.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "blurtool.h"
-#include <QPainter>
+#include <QApplication>
 #include <QGraphicsBlurEffect>
 #include <QGraphicsPixmapItem>
 #include <QGraphicsScene>
-#include <QApplication>
+#include <QImage>
+#include <QPainter>
 
-BlurTool::BlurTool(QObject *parent) : AbstractTwoPointTool(parent) {
+BlurTool::BlurTool(QObject *parent) : AbstractTwoPointTool(parent) {}
 
+QIcon BlurTool::icon(const QColor &background, bool inEditor) const
+{
+  Q_UNUSED(inEditor);
+  return QIcon(iconPath(background) + "blur.svg");
 }
+QString BlurTool::name() const { return tr("Blur"); }
 
-QIcon BlurTool::icon(const QColor &background, bool inEditor) const {
-    Q_UNUSED(inEditor);
-    return QIcon(iconPath(background) + "blur.svg");
-}
-QString BlurTool::name() const {
-    return tr("Blur");
-}
+QString BlurTool::nameID() { return QLatin1String(""); }
 
-QString BlurTool::nameID() {
-    return QLatin1String("");
+QString BlurTool::description() const
+{
```

Comment:
Same line fan here for collapsing reasons :)

Reply:
Yes I agree. I submitted a .clang-format file to keep the formatting consistent for the project. In the future I want to add a check to a github action to make sure the code conforms. 

I am happy to change the .clang-format to what ever style you want, the most important thing is to stay consistent once we decide.

Reply:
I'll post this in slack also, but we can vote on which standard clang-format profile to use instead of my personal preference: https://www.strawpoll.me/20877923

Reply:
Wouldn't it be better to offer Blur and Pixelate? By the way, the source of this PR could be easier if you just scale down and enlarge the image, this creates the same effect. In my case it looks like this, it only lacks to vary the intensity:
```
void BlurTool::process( QPainter &painter, const QPixmap &pixmap, bool recordUndo )
{
    if ( recordUndo )
    {
        updateBackup( pixmap );
    }

    QPoint &p0      = m_points.first;
    QPoint &p1      = m_points.second;
    QRect selection = QRect( p0, p1 ).normalized();

    int width = selection.width() * 0.1;

    QPixmap t = pixmap.copy( selection );
    t         = t.scaledToWidth( qMax( width, 10 ), Qt::SmoothTransformation );
    t         = t.scaledToWidth( selection.width() );
    painter.drawImage( selection, t.toImage() );
}
```

Reply:
One thing that was requested in the initial Issue was the ability to adjust the pixel width. If this simple method is able to do that I can change it. 

I did not want to overload the user with choices and since blur and pixelate are so similar I replaced blur with the pixelate algorithm.

Reply:
> One thing that was requested in the initial Issue was the ability to adjust the pixel width. If this simple method is able to do that I can change it.

I think the exact pixel size could only be determined approximately, but it would be no problem to add an intensity of 1-100. From low intensity to a pixelation where everything is just a single colored object.

> I did not want to overload the user with choices and since blur and pixelate are so similar I replaced blur with the pixelate algorithm.

In this case it is probably better to rename the whole class including all files.

Reply:
I'm going to update a separate issue for renaming everything. I really want to get this merged so everyone can work off the same base, but I agree to change the name and class name.

Reply:
The calculation of the intensity could look like this:
`int width = selection.width() * ( 0.5 / qMax( 1, m_thickness ));`

As an example here is a photo of how different intensity of the pixelation looks like with this algorithm.
![Example](https://user-images.githubusercontent.com/39984895/92219382-c7a3da80-ee9a-11ea-8b42-1378935126b3.png)